portfolio/src/main/webapp/scripts/utilities.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Ensures that the stated arguments are present. 
+ * @param {...(NodeList|HTMLCollection|HTML element|string|Array)} args
+ *    Arguments to be checked.
+ */
+function ensureNonNull(... args) {
+  /**
+   * Indicates error type.
+   * @enum {string}
+   */
+  const ErrorType = {
+    NULL_VALUE: 'Null value',
+    EMPTY_STR: 'Empty string',
+    EMPTY_COLL_OR_LIST: 'Empty HTMLCollection/NodeList',
+    EMPTY_ARR: 'Empty array'
+  };
+  Object.freeze(ErrorType);
```

Comment:
You can simply do
```
const ErrorType = Object.freeze({
    NULL_VALUE: 'Null value',
    EMPTY_STR: 'Empty string',
    EMPTY_COLL_OR_LIST: 'Empty HTMLCollection/NodeList',
    EMPTY_ARR: 'Empty array'
  });
```
Because you may not want people to easily mutate the enum before you freeze it, so it's we'd better make sure that it's properly freezed at time of declaration.

Reply:
This is true! Thanks (: